comment_scraper.py

```diff
@@ -1,87 +1,96 @@
+            versions = client.get_asset_children(asset['id'])
+            for v_asset in versions:
+                comments = client.get_comments(v_asset['id'])
+                for comment in comments:
+                    comment['parent_id'] = v_asset['parent_id']
+                    comment['name'] = v_asset['name']
+                    comment_list.append(comment)
 
-# Saving the asset name and parent ID.
+    return comment_list
 
-            asset_name = asset['name']
-            parent_id = asset['parent_id']
-            vfiles = client.get_asset_children(asset['id'])
 
-            for asset in vfiles.results:
-                asset_name = asset['name']
-                parent_id = asset['parent_id']
-                if asset['type'] == "file":
-                    if asset['comment_count'] > 0:
-                        comments = client.get_comments(asset['id'])
-                        my_comment_list = [comment for comment in comments.results]
-                        for object in my_comment_list:
-                            object.update({'parent_id':parent_id})
-                            object.update({'name':asset_name})
-                        comment_list.append(my_comment_list)
+def flatten_dict(d):
+    # The get_comments API response is verbose and contains nested objects.
+    # Use this helper functon to flatten the dict holding the API response data
+    # and namespace the attributes.
 
-# Takes a root asset ID for a project and a developer token.
-# Returns a comment list with all assets.
+    def expand(key, val):
+            if isinstance(val, dict):
+                return [ (key + '.' + k, v) for k, v in flatten_dict(val).items() ]
+            else:
+                return [ (key, val) ]
+    
+    items = [ item for k, v in d.items() for item in expand(k, v)]
 
-def get_all_project_comments(root_asset_id, token):
-    comment_list = []
-    client = FrameioClient(token)
+    return dict(items)
 
-    all_comments(client, root_asset_id, comment_list)
 
-    return comment_list
+def write_comments_csv(c_list):
+    # Writes comments to comments.csv
+    # Any attributes you add to the headers list will automatically be written to the CSV
+    # The API returns many attributes so familiarize with the response data!
+    headers = ['text', 'name', 'inserted_at', 'timestamp', 'has_replies', 'parent_id', 'owner.name', 'owner.email', 'owner_id', 'owner.account_id']
 
-# Get all the comments on the project you choose by providing a root asset ID and a developer token.
+    # Flattening the comments dicts is not at all necessary, but the namespacing
+    # makes the CSV headers much more readable.
+    flat_comments_list = []
+    for c in c_list:
+        flat_comments_list.append(flatten_dict(c))
 
-responses = get_all_project_comments(ROOT_ASSET_ID, TOKEN)
+    with open('comments.csv', 'w') as file:
+        f_csv = csv.DictWriter(file, headers, extrasaction='ignore')
```

Comment:
interesting. so _this_ is where all the extra keys in the dict get dropped. that's kind of a bummer. i think if you wanted to make this code a little more flexible / reusable, the thing to do would be to have a default list of keys (the headers list above) and drop all of the other keys before coming to the csv writer. that way your writer is _just_ writing, it's not also doing this vertical filtering thing. certainly more work the way i'm describing, but also would allow someone to call into your function and say "i want these other fields and i want you to send it somewhere other than a flat csv file." 

just something to think about i suppose.